### DISCUSSION TITLE
Bug in custom fields

### DISCUSSION THREAD

**Discussion by saman222** [joomengine/Joomla-Component-Builder#671] [Extention Issues]
Hi
I was created an custom field with some PHP code in it. (mycusotmfield)
after compiling and installing the component when I want to change php part of field , no change is applied
and changes just applied when I change types value field in property of custom field (mycustomfieldone)
its ok but when I change type of field to the first type, php field that writes in admin/component/my_component/models/fields/mycustomfield.php backs to the first PHP that I wrote in first step
This problem persists even with the deletion of mycustomfield.php and mycustomfieldone.php

**Comment by TLWebdesign** [comment]
Hi,
Could you check that there is no item in the trash with the same types value? I had this issue the other day where there were trashed items with the same "system name" and it cause those to be picked up instead of the active version with that name. I wonder if this is the same problem here.

**Comment by Llewellynvdm** [comment]
if you have more then one custom field with the same field_type "mycusotmfield" then JCB will use the first it gets to set the PHP. Now since we often have the issue that we want to force one to be the **PRIME/MAIN** field which JCB will use we added a switch called "Prime PHP"

![image](https://user-images.githubusercontent.com/5607939/108131817-899e1a00-70ba-11eb-8c21-af5e127ff3f0.png)

Selecting that option in custom fields, and setting it to 1 will force that field to be used as the **PRIME**, unless you also have the same switch set on the other field with the same **field_type**. So **only one** of these fields should be set to prime, and then that fields php will be used instead of any of the other fields that may have different names, but the same "custom" field_type.

This is what the prime_php switch description says:
```text
This field makes sure that the PHP used here is used to build the field type, and other are custom fields with the same field type are ignored. So to avoid that they over write the PHP added here. You should only have one prime per/type. To disable remove the field or set to 0
```
